backend/routes/projects.js

```diff
@@ -30,4 +52,63 @@ router.get("/", [authorizeUser(["roster"])], (req, res, next) => {
       });
 });
 
+/**
+ * Returns a specific project given its id.
+ */
+router.get("/:project_id", [authorizeUser(["roster"])], (req, res, next) => {
+  getOneProject(req.params.project_id)
+    .then((project) => {
+      if (project === null)
+        res.status(400).json({ message: "Project with the given id doesn't exist" });
+      else res.status(200).json({ project });
+    })
+    .catch((err) => {
+      next(err);
+    });
+});
+
+/**
+ * Creates a new project given that the user has permissions to
+ * create projects.
+ */
+router.post(
+  "/",
+  [
+    validators.name,
+    validators.description,
+    validators.client,
+    validators.phase,
+    validators.timeline_quarter,
+    validators.timeline_year,
+    validators.project_manager,
+    validators.outreach,
+    validators.designers,
+    validators.developers,
+    validators.file_link,
+    validators.file_name,
+    validateRequest,
+    authorizeUser(["roster", "project_create"]),
+  ],
+  (req, res, next) => {
+    addProject({
```

Comment:
Is there a reason why `req.body` isn't passed in directly here?

Reply:
In case extra attributes are passed in the JSON. I followed it because the same was there for the POST user route.

Reply:
Okay, that's fair! I don't think it's strictly necessary, because [Mongoose ignores unrecognized fields by default](https://mongoosejs.com/docs/guide.html#strict), but I suppose it doesn't hurt to leave it in either.